tests/basic_test.go

```diff
@@ -191,9 +190,79 @@ routes:
 	})
 }
 
-func resetTimeStamp(s types.GeneratedState) types.GeneratedState {
-	s.MetaInfo.TimeStamp = time.Time{}
-	return s
+func testGeneratedStateTimestampsAfterOneSecond(t *testing.T) {
```

Comment:
1. Can you please simplify the test to a minimum need to get the metadata? All other details hide the main intent of the test.
2. I think that the test should just call `nmpolicy.GenerateState` twice, and make sure one is greater than the other.

WDYT?

Reply:
No need for call it twice.

Reply:
I think the way it is done now is more complex, but fine.
Just one nit: Remove the `AfterOneSecond` from the name.
You could name it: `testGenerateUniqueTimestamps`